src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,96 @@
+import { ChangeEvent, useEffect, useRef } from 'react';
+import { Todo } from '../../types/Todo';
+import { useDispatch, useGlobalState } from '../../GlobalStateProvider';
+import { Type } from '../../types/Action';
+import { addTodos, USER_ID } from '../../api/todos';
+import { ErrorType } from '../../types/Errors';
+
+type Props = {
+  handleError: (message: string) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ handleError }) => {
+  const { title, isSubmitting, editingId, updatingId } = useGlobalState();
+  const dispatch = useDispatch();
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const addTodo = ({
+    userId,
+    title: newTitle,
+    completed,
+  }: Omit<Todo, 'id'>) => {
+    dispatch({ type: Type.setErrorMessage, payload: '' });
+    dispatch({ type: Type.setIsSubmitting, payload: true });
+
+    return addTodos({ userId, title: newTitle, completed })
+      .then(newTodo => {
+        dispatch({ type: Type.AddTodo, payload: newTodo });
+        dispatch({ type: Type.setTitle, payload: '' });
+      })
+      .catch(() => {
+        handleError(ErrorType.ADD_TODO);
+      })
+      .finally(() => {
+        dispatch({ type: Type.setIsSubmitting, payload: false });
+        dispatch({ type: Type.setTempTodo, payload: null });
+      });
+  };
+
+  useEffect(() => {
+    if (inputRef.current && !editingId && !updatingId) {
```

Comment:
only one input can be in focus at the time. editingId is defined only when editing input exists (and it exists only when it's in focus).  So there is no need to check it here. I.e. it could be just the local state inside the TodoItem component.

Not critical, just FYI 🙂

Reply:
Thank you very much for the review and comments! Just checked although - it doesn't work without editingId 🤷‍♀️